crates/oracle/src/oracle.rs

```diff
@@ -71,20 +73,17 @@ impl Oracle {
 
     /// Checks if the Subgraph should consider that the Subgraph is at a previous epoch compared to
     /// the Epoch Manager.
-    async fn detect_new_epoch(&mut self) -> Result<bool, Error> {
-        // Before anything else, we must get the latest subgraph state
-        debug!("Querying the subgraph state...");
-        self.subgraph_state.refresh().await;
-        self.subgraph_state.result()?;
-
+    async fn detect_new_epoch(&self, subgraph_state: &SubgraphState) -> Result<bool, Error> {
         // Then we check if there is a new epoch by looking at the current Subgraph state.
-        let last_block_number_indexed_by_subgraph = match self.is_new_epoch().await? {
+        let last_block_number_indexed_by_subgraph = match self.is_new_epoch(subgraph_state).await? {
             // If the Subgraph is uninitialized, we should skip the freshness and epoch check
             // and return `true`, indicating that the Oracle should send a message.
             //
             // Otherwise this could lead to a deadlock in which the Oracle never sends any
             // message to the Subgraph while waiting for it to be initialized.
-            NewEpochCheck::SubgraphIsUninitialized => return Ok(true),
+            NewEpochCheck::SubgraphIsUninitialized {
+                subgraph_latest_indexed_block,
+            } => subgraph_latest_indexed_block,
```

Comment:
This got me thinking.

If the subgraph isn't initialized, then there is nothing left for the Oracle to do, as it doesn't send `RegisterNetworks` anymore.

I think that we could treat this case as a recoverable error, wdyt?

Reply:
Oh my, you're right. There's absolutely nothing for the oracle to do!

Yes, a recoverable error sounds like the best idea. Will get started right away.